letta/client/client.py

```diff
@@ -2188,48 +2188,8 @@ def delete_human(self, id: str):
     # tools
 
     # TODO: merge this into create_tool
-    def add_tool(self, tool: Tool, update: Optional[bool] = True) -> Tool:
-        """
-        Adds a tool directly.
-
-        Args:
-            tool (Tool): The tool to add.
-            update (bool, optional): Update the tool if it already exists. Defaults to True.
-
-        Returns:
-            None
-        """
-        if self.tool_with_name_and_user_id_exists(tool):
-            if update:
-                return self.server.update_tool(
-                    ToolUpdate(
-                        id=tool.id,
-                        description=tool.description,
-                        source_type=tool.source_type,
-                        source_code=tool.source_code,
-                        tags=tool.tags,
-                        json_schema=tool.json_schema,
-                        name=tool.name,
-                    ),
-                    self.user_id,
-                )
-            else:
-                raise ValueError(f"Tool with id={tool.id} and name={tool.name}already exists")
-        else:
-            # call server function
-            return self.server.create_tool(
-                ToolCreate(
-                    id=tool.id,
-                    description=tool.description,
-                    source_type=tool.source_type,
-                    source_code=tool.source_code,
-                    name=tool.name,
-                    json_schema=tool.json_schema,
-                    tags=tool.tags,
-                ),
-                user_id=self.user_id,
-                update=update,
-            )
+    def create_or_update_tool(self, tool_create: ToolCreate) -> Tool:
```

Comment:
maybe unpack the arguments here to avoid requiring a `ToolCreate`?

Reply:
factored this out, deleted this exposed endpoint